apps/obsidian/src/services/QueryEngine.ts

```diff
@@ -0,0 +1,79 @@
+import { TFile, App } from "obsidian";
+import { getAPI } from "obsidian-dataview";
+
+export type SearchNodeOptions = {
+  excludeFile?: TFile;
+  minQueryLength?: number;
+};
+
+export class QueryEngine {
+  private dv: any;
+
+  constructor(app: App) {
+    this.dv = getAPI(app);
```

Comment:
`this.dv` isn't initializing for me.
![image](https://github.com/user-attachments/assets/0421ff47-8b99-4746-b1d7-846ab4c38781)

My suggestions would be to change this to a utility function (`utils/queryEngine.ts` or individual `utils/searchNodeByTitle.ts`).  Unless you have a reason that a class would be better.